### PR TITLE
Debugging and fixing srcs issues 24 and 25

### PR DESCRIPTION
**Pull request by neomatrix369**
The fix was a simple one although wouldn't have immediately guessed it.

Had to reproduce the issue via tests, although the two issues similar in nature have different setups. Happy to take feedback on them and fix them where needed. Possibly there are better idioms from the project itself that can be used to make the tests simpler.

All other tests are passing and green on the CI.

Also ran the two .srcs files from issues #24 and #25 and with the fix they work fine without any array-index-out-of-bounds exceptions.

### CODE REVIEW

**Comment by neomatrix369** [comment]
Code coverage says its only 1% more complex ;) - what do you reckon?

**Comment by julianghionoiu** [comment]
> Code coverage says its only 1% more complex ;) - what do you reckon?

Not concerned about complexity.

**Comment by julianghionoiu** [comment]
I have reviewed the PR and I am pleased to see that you got to the bottom of the issue.

It is good that you have captured the issue with tests, but those are user tests and are hard to understand an maintain.

The next step is to boil down those tests into one e2e test that is using the framework from ConvertToGitEdgeCasesTest.java. This will prove that we have an understanding of how it can be reproduced and what sort of newline characters it affects (Windows CRLF?)

So I am now expecting for the two tests to transform into one similar to ConvertToGitEdgeCasesTest.java

File: src/main/java/tdl/record/sourcecode/snapshot/file/ToGitConverter.java [hunk 1]
```diff
@@ -61,8 +61,8 @@ private void processSegment(Reader reader) throws Exception {
         try {
             Header header = reader.getFileHeader();
             Segment segment = reader.nextSegment();
-            writeDirFromSnapshot(segment);
             listener.commitSegment(segment);
+            writeDirFromSnapshot(segment);
```
Comment (julianghionoiu) [inline src/main/java/tdl/record/sourcecode/snapshot/file/ToGitConverter.java:64]: What is the reason for this change?

**Review by julianghionoiu — APPROVED** [review]

File: src/main/java/tdl/record/sourcecode/snapshot/file/ToGitConverter.java (hunk 1 above)
Comment (neomatrix369) [reply src/main/java/tdl/record/sourcecode/snapshot/file/ToGitConverter.java]: I have out that in the commit message - previously when `writeDirFromSnapshot(segment);` we wouldn't know at which time segment it failed and also was a red-herring when investigating the faulty segments.

By switching over, we always know which segment is being processed, and if it fails we know which one to look for, it is in the logs or on the console - no debugging needed.

Comment (neomatrix369) [reply src/main/java/tdl/record/sourcecode/snapshot/file/ToGitConverter.java:64]: I have explained that in the commit message (99a1847389b2) - previously when `writeDirFromSnapshot(segment);` we wouldn't know at which time segment it failed and also was a red-herring when investigating the faulty segments.

By switching over, we always know which segment is being processed, and if it fails we know which one to look for, it is in the logs or on the console - no debugging needed.
